### DISCUSSION TITLE
How to use api v2?

### DISCUSSION THREAD

**Discussion by sudoexec** [usememos/memos#2657] [Q&A] [answered]
Try to get memos via api v2(/api/v2/memos), but got error:
```
{
"code": 14,
"message": "connection error: desc = \"transport: Error while dialing: dial tcp :5231: connect: connection refused\"",
"details": []
}
```
Is api v2 available?

**Comment by jhademcconnell** [comment]
@sudoexec it looks like the v2 endpoints may not be exposed yet. To call the REST API that retrieves all memos for your instance in the v1 endpoints, you would do:

Request:
`{host name}/api/v1/memo/all`

Response:
`
[
  {
    "id": 68,
    "rowStatus": "NORMAL",
    "creatorId": 1,
    "createdTs": 1703599610,
    "updatedTs": 1703599610,
    "displayTs": 1703599610,
    "content": "Example Memo",
    "visibility": "PUBLIC",
    "pinned": false,
    "parent": null,
    "creatorName": "",
    "creatorUsername": "",
    "resourceList": [],
    "relationList": []
  }
]
`
You may have to wait until a release confirms the v2 endpoints are ready to be called. There usually is a swagger.yaml file but I don't see it in the v2 API folder like I do in v1 API.

**Reply by sudoexec** [reply]
Thanks. I know v1 is there, but v1 doesn't have a timestamp filter and v2 does.

**Comment by boojack** [comment, marked as answer]
Fixed in https://github.com/usememos/memos/commit/0dfc367e566ff8e15fe4cc7558ce691de755589e. Now it's worked. e.g., https://demo.usememos.com/api/v2/memos
